pinot-broker/src/main/java/org/apache/pinot/broker/queryquota/HelixExternalViewBasedQueryQuotaManager.java

```diff
@@ -323,11 +322,34 @@ public void processQueryQuotaChange() {
       }
       int onlineBrokerCount = otherOnlineBrokerCount + 1;
 
+      // Get stat from property store
+      String tableConfigPath = constructTableConfigPath(rawTableName, tableType);
+      Stat stat = propertyStore.getStat(tableConfigPath, AccessOption.PERSISTENT);
+      if (onlineBrokerCount == queryQuotaConfig.getNumOnlineBrokers() && stat.getVersion() == queryQuotaConfig
+          .getTableConfigStatVersion()) {
+        LOGGER.info(
+            "Number of brokers remains the same. Table config remains the same. No need to update quota for Table {}.",
+            tableNameWithType);
+        continue;
+      }
+
+      // Get latest quota config for table.
+      QuotaConfig quotaConfig = getQuotaConfigFromPropertyStore(propertyStore, rawTableName, tableType);
```

Comment:
You don't need to fetch a new quota config if version matches but numOnlineBrokers changed

Reply:
Good idea. 👍 Done.